### PR TITLE
140 secp signing and identity chain persistence

### PR DESCRIPTION
**Pull request by zupzup**
## 📝 Description

Draft for feedback on identity chain persistence structure. Tests are missing. The implementation of the different identity chain use-cases will be in a different PR.

This implements two things:
1. It changes the signature/verification mechanism within the blockchain implementations to use the secp256k1 keys, instead of RSA
2. It implements a persistence structure for storing the identity blockchain

Relates to #140 

---

## ✅ Checklist

Please ensure the following tasks are completed before requesting a review:

- [x] My code adheres to the coding guidelines of this project.
- [x] I have run `cargo fmt`.
- [x] I have added or updated tests (if applicable).
- [x] All CI/CD steps were successful.
- [x] I have updated the documentation (if applicable).
- [x] I have checked that there are no console errors or warnings.
- [x] I have verified that the application builds without errors.
- [x] I've described the changes made to the API. (modification, addition, deletion).

---

## 🚀 Changes Made

- **New Features:**
  - Sign/Verify using Secp256k1
  - Persistence structure for identity blockchain

---

## 💡 How to Test

Run tests.
Upon identity creation, the identity blockchain should have been created in SurrealDB

---

## 🤝 Related Issues

List any related issues, pull requests, or discussions:

- #140

---

## 📋 Review Guidelines

Please focus on the following while reviewing:

- [ ] Does the code follow the repository's contribution guidelines?
- [ ] Are there any potential bugs or performance issues?
- [ ] Are there any typos or grammatical errors in the code or comments?

### CODE REVIEW

File: src/blockchain/identity/mod.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    fn blocks_mut(&mut self) -> &mut Vec<Self::Block> {
+        &mut self.blocks
+    }
+}
+
+impl IdentityBlockchain {
+    /// Creates a new identity chain, encrypting the identity with the public rsa key
+    pub fn new(identity: &Identity, keys: &BcrKeys, timestamp: i64) -> Result<Self> {
+        let identity_bytes = to_vec(&identity)?;
+        let genesis_hash = hex::encode(&identity_bytes);
```
Comment (mtbitcr) [inline src/blockchain/identity/mod.rs:119]: can we already use base64?

**Review by mtbitcr — APPROVED** [review]
lgtm

File: src/blockchain/identity/mod.rs (hunk 1 above)
Comment (zupzup) [reply src/blockchain/identity/mod.rs:119]: I will add it in the next PR when I implement the operations.
I thought about an API like adding `data` and `data_encoded` and maybe even `data_decrypted` to the trait, so we can transition the bill code later easily as well.

Comment (mtbitcr) [reply src/blockchain/identity/mod.rs:119]: Hmm. Let me think about. Let's discuss tomorrow

**Pull request merged**